### ISSUE TITLE
iTunesDB podcasts - questionable amount

### ISSUE THREAD

**Issue by AZProductions** [raleighlittles/iTunesDB-Parser#1]
Do you need an ItunesDB file with music metadata inside of it? I could provide you one with that.

**Comment by raleighlittles** [comment]
@AZProductions  Yes! That would be much appreciated, thank you.

**Comment by AZProductions** [comment]
Here you go, I called the Ipod 'Dev Ipod', so that might show up internally. [iTunesDB with music.zip](https://github.com/raleighlittles/iTunesDB-Parser/files/11866529/iTunesDB.with.music.zip)

**Comment by AZProductions** [comment]
I also have other files like ItunesControl, let me know if you need it.

**Comment by raleighlittles** [comment]
@AZProductions Great, thank you again for this. I've started a new branch where I'll be adding support for parsing the music file.

The other files would be nice to have too, but it looks like iTunesControl doesn't really have anything useful in it: http://www.ipodlinux.org/ITunesDB/#iTunesControl_File so we can probably skip that one specifically.

**Comment by raleighlittles** [comment]
@AZProductions I finally got around to decoding the iTunesDB file. 

Here's a CSV file with all of the songs that I could decode. Could you review this and make sure it mostly looks right?

[2023-06-26_itunes_db.csv](https://github.com/raleighlittles/iTunesDB-Parser/files/12063454/2023-06-26_itunes_db.csv)

I was having an issue with some of the timestamps being empty. I think it might've been that some of the songs on your iPod weren't downloaded from iTunes, and were instead self-uploads -- is that correct?

Also, besides songs, did you have any other media on your iPod, like Podcasts, Movies, TV shows, etc? My code didn't detect anything other than songs, and I don't know if it's because there's a bug or because the iPod really only had songs on it.

**Comment by AZProductions** [comment]
Yes, that's definitely possible, I used a 3rd party app to add some songs. (Although I did that by the album, not per song. I always completely uploaded albums or artists through iTunes/3rd party apps.) Also, the iPod only had songs on it, so if you want a version with podcast/movie, I'll have to set that up.

Thanks!

**Comment by raleighlittles** [comment]
@AZProductions Ah, okay, that could explain part of it.

 If you could add Podcasts and then re-upload the iTunesDB file, that would be much appreciated. In the meantime, if you upload [any of the other filetypes](https://github.com/raleighlittles/iTunesDB-Parser#about), I can work on adding support for those while you get the podcasts setup?

**Comment by raleighlittles** [comment]
@AZProductions Hey just wanted to follow up, if you still have any of the other iTunesDB files, can you add them here?

**Comment by AZProductions** [comment]
Hello, sorry for the late reply. I'm currently setting up a new ipod with podcasts playlists etc, will give you the files soon!

**Comment by AZProductions** [comment]
[iPod_Control.zip](https://github.com/raleighlittles/iTunesDB-Parser/files/12565661/iPod_Control.zip)
Here's the full iPod_Control folder including 3 podcasts by the same person and more songs. I also rated a couple of them btw

**Comment by raleighlittles** [comment]
@AZProductions I saved the podcast parsing for last, but I started looking at the other files, right now I'm on the last one, the `Preferences` file, but I'm having trouble matching up the documentation with the actual file contents.

What timezone is your iPod in?

**Comment by AZProductions** [comment]
GMT+1 (Amsterdam)

**Comment by thomas-louvigne** [comment]
hi, i can also send you  photo and data.
But i don't want to put it on the web, can i send it to you to a mail ?
I loved to get back the photo of the ipod and i don't know how to do it.

**Comment by raleighlittles** [comment]
@thomas-louvigne thanks for reaching out. I don't currently have support for extracting actual thumbnail images yet unfortunately, that is the next feature I will be doing. Once that feature is ready I'll notify you. Please see [the other issue about this](https://github.com/raleighlittles/iTunesDB-Parser/issues/4).

**Comment by raleighlittles** [comment]
@AZProductions I only found 1 podcast on the iTunesDB file for some reason. Here's a screenshot of its info -- its titled "Can You Start a YouTube Channel in 2023?". I wonder if it's because this was the only podcast that was "downloaded" (?). If you're sure there's 3 podcasts on that iPod, I can't think of why I'm only seeing that one -- nothing in the iPodLinux documentation would explain why unfortunately.
![2023122_podcasts-csv](https://github.com/raleighlittles/iTunesDB-Parser/assets/2137201/e381748e-c87c-43ab-93a6-7a3acc65c8b2)

**Comment by raleighlittles** [comment]
Closing due to inactivity, don't have enough examples of iTunesDB files with podcasts in them to be able to reliably diagnose this better
